tests/kernel/queue/src/test_queue_fail.c

```diff
@@ -72,16 +74,19 @@ void test_queue_append_list_error(void)
 			"failed to CHECKIF tail == NULL");
 	/* Initializing the queue for re-using below */
 	k_queue_init(&queue);
+
+	/* Append unnormal list(just one node)into the queue for sub-thread */
+	head = &data_l[0];
+	head->snode.next = NULL;
+
 	k_thread_create(&tdata, tstack, STACK_SIZE, tThread_entry, &queue,
-			NULL, NULL, K_PRIO_PREEMPT(0), 0, K_NO_WAIT);
+			head, NULL, K_PRIO_PREEMPT(0), 0, K_NO_WAIT);
 	/* Delay for thread initializing */
-	k_sleep(K_MSEC(500));
```

Comment:
Hi @Zhaoningx, I think if we want change the timing for delay, is there any risk to effect the physical boards?

Reply:
Hi @enjiamai, thanks for your comment, I have re-designed this case. Could you help to review it?